docs/build/js-client/10-marine-js.md

```diff
@@ -1,63 +1,138 @@
+The lib should do some simple processing locally, e.g., directly in a browser and also parallel the work if necessary.
 
-Locates a WASM file in the specified npm package and loads it. The function accepts two arguments:
+### Summary
 
-- Name of an npm package
-- Path to a WASM file relative to the npm package
+These use cases have something in common—they need to host a logic locally.
+Moreover, sometimes writing a logic in pure JavaScript is not feasible.
+For example, AI processing logic would be a lot effective if written in low-level language like Rust
 
-This function can only be used in nodejs. Trying to call it inside a browser will result in an error.
+## Use cases
 
-**loadWasmFromServer**
+Usually you don't need to interact with Marine JS directly except the following:
+- Cross-platform service which works both on Nox peer and JS client peer
+- Heavy computations, CPU intensive tasks
+- Would prefer to write service in Rust or other language with compilation to WASM
 
-Loads a WASM file from the service hosting the application. It accepts the file path on server and returns a buffer compatible with the `FluencePeer` API.
+In any matched case, writing your service as WASM more suitable than doing so in JavaScript.
 
-:::info
-The function will try to load a file into SharedArrayBuffer if the site is cross-origin isolated. Otherwise, the return value falls back to Buffer, which is less performant but is still compatible with the `FluencePeer` API.
+## What is Marine JS
 
-We strongly recommend to set-up cross-origin headers. For more details, see the [MDN page](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer#security_requirements).
-:::
+When JS client starts up, it starts with initialization Marine JS runtime.
+The runtime hosts wasm services and even allows you to register your own wasm services.
+For example, AquaVM service for processing particles and Air code resides in Marine JS.
+That's why Marine JS is a foundation of JS client.
 
-This function can only be used in a browser. Trying to call it inside nodejs will result in an error.
+## Extending JS client with Marine services
 
-### Registering services in FluencePeer
+> Currently only [pure](https://fluence.dev/docs/build/glossary#pure-module) single-module services are supported.
 
-After the file has been loaded, it can be registered in `FluencePeer`. To do so, use the `registerMarineService` function.
+Using Marine services is pretty straight forward. The first thing to do is to load a compiled WASM file into your environment.
 
-To remove a service, use the `registerMarineService` function.
+### Loading module binary
 
-You can pick any unique service id. Once the service has been registered it can be referred in Aqua code by the specified id. For example:
+Here are the examples of how to load your wasm file in browser or node
```

Comment:
For stylistic reasons I suggest always formatting the name like this `Node.js` cause that's how this is formatted officially. This will also remove any confusion with any other notion of `node` that can appear in conversation 
```suggestion
Here are the examples of how to load your wasm file in browser or Node.js
```